### PR TITLE
Optimize buildtime

### PR DESCRIPTION
**Pull request by johangirod**
Before : 
screen shot 2017-03-31 at 20 52 41

After: 
screen shot 2017-03-31 at 20 47 41

😗 ⚡️

### CODE REVIEW

File: config/build/webpack.config.common.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import { HotModuleReplacementPlugin, LoaderOptionsPlugin, NamedModulesPlugin } from 'webpack';
+import { HotModuleReplacementPlugin, LoaderOptionsPlugin, NamedModulesPlugin,
+    } from 'webpack';
```
Comment (christophehurpeau) [inline config/build/webpack.config.common.js]: ?

File: config/build/webpack.config.vendor.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import mergeWith from 'lodash.mergewith';
+import webpack from 'webpack';
+import path from 'path';
+import { createBabelLoader, config } from './webpack.config.common.js';
+import { concat, appResolve, appModules } from '../utils';
+
+// eslint-disable-next-line
+const appDependencies = require(appResolve('package.json')).dependencies;
+const dependancies = {
```
Comment (christophehurpeau) [inline config/build/webpack.config.vendor.js]: are we using american or british english ? in american, dependencies are spelled with an `e`, not an `a` :)

File: config/build/webpack.config.vendor.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    'react-router-redux': true,
+    'isomorphic-style-loader': true,
+    'react-helmet': true,
+    'redbox-react': true,
+    'redux-thunk': true,
+    ...appDependencies,
+};
+const vendors = Object.keys(dependancies)
+    .filter(vendor => !vendor.startsWith('koa-proxy'))
+    .filter(vendor => vendor !== 'co-request'); /* WTF */
```
Comment (christophehurpeau) [inline config/build/webpack.config.vendor.js]: can you be a bit more explicit than WTF ?

File: src/server/components/App.jsx [hunk 4]
```diff
@@ -15,15 +14,18 @@ const propTypes = {
     }).isRequired,
     insertCss: PropTypes.func.isRequired,
     renderProps: PropTypes.objectOf(PropTypes.any).isRequired,
-    mainEntry: PropTypes.string.isRequired,
+    entries: PropTypes.arrayOf(PropTypes.shape({
+        src: PropTypes.string.isRequired,
+        async: PropTypes.bool.isRequired,
```
Comment (christophehurpeau) [inline src/server/components/App.jsx]: why is it required ?

File: config/build/webpack.config.common.js [hunk 5]
```diff
@@ -34,7 +35,7 @@ export const createResolveConfigLoader = () => ({
 
 export function config(options) {
     return {
-        devtool: options.dev && 'source-map',
+        devtool: options.dev && 'eval-cheap-module-source-map',
```
Comment (christophehurpeau) [inline config/build/webpack.config.common.js]: eval-cheap-module doesn't work well...

Comment (johangirod) [inline config/build/webpack.config.common.js]: Aww :(
You have the link to the corresponding issue ? We could at least link it to here / upvote it.

Comment (christophehurpeau) [inline config/build/webpack.config.common.js]: I tested them all for pobpack-browser and the only one working with chrome and firefox and react-redbox was source-map

Comment (johangirod) [inline config/build/webpack.config.common.js]: Well... I guess I'm going to revert it then. The good news is that it's only a small part of the performance improvement.

Comment (alepee) [inline config/build/webpack.config.common.js]: it seems @christophehurpeau _"pèse dans le game"_

**Comment by johangirod** [comment]
Revert, while waiting for #332 to be fixed (work in progress)

**Pull request merged**